proto/injective/exchange/v2/query.proto

```
@@ -1387,6 +1440,50 @@ message QueryCrossMarginPoolSnapshotResponse {
   // Field 16 removed: canonical_liquidation_target_market_id is no longer
   // relevant — MsgLiquidateCrossMarginPool closes all positions atomically.
   reserved 16;
+
+  // Ratcheted maintenance margin total: max(maintenance at current mark,
+  // maintenance at entry notional) summed across positions. This is the
+  // maintenance floor used in the cash-floor and withdrawable formulas, and is
+  // distinct from maintenance_margin_total. It is NOT derivable from the other
+  // response fields, so off-chain monitors need it to reconstruct the chain's
+  // admission / withdrawal predicates.
+  string ratcheted_maintenance_margin_total = 17 [
+    (gogoproto.customtype) = "cosmossdk.io/math.LegacyDec",
```

Buf (1.71.0)

[error] cannot find `gogoproto.customtype` in this scope

(COMPILE)